mint-nft/index.js

```diff
@@ -0,0 +1,178 @@
+  const content = readFileSync(path.join(__dirname, './data/nft.json'));
+  const data = JSON.parse(content);
+  if (!data || !data.name) throw new Error('Invalid NFT data json');
+  console.log(`Creating NFT Class - ${data.name}`);
+
+  let classConfig = null;
+  if (nftMaxSupply) classConfig = { nftMaxSupply };
+
+  let { uri } = data;
+  const isUriHttp = uri && uri.startsWith('https://');
+  if (isUriHttp) uri = addParamToUrl(uri, { iscn_id: iscnId });
+  const res = await signingClient.createNFTClass(
+    account.address,
+    iscnId,
+    {
+      ...data,
+      uri,
+    },
+    classConfig,
+  );
+  console.log(`Creating NFT Class - Completed ${res.transactionHash}`);
+  const rawLogs = JSON.parse(res.rawLog);
+  const event = rawLogs[0].events.find(
+    (e) => e.type === 'likechain.likenft.v1.EventNewClass',
+  );
+  const attribute = event.attributes.find((a) => a.key === 'class_id');
+  const classId = ((attribute && attribute.value) || '').replace(/^"(.*)"$/, '$1');
+  console.log(`Creating NFT Class - NFT Class ID: ${classId}`);
+  return classId;
+}
+
+async function mintNFTsFromJSON(classId, nftCount, signingClient, account) {
+  console.log('Minting NFTs - Reading data from ./data/nft.json...');
+  const content = readFileSync(path.join(__dirname, './data/nft.json'));
+  const data = JSON.parse(content);
+  if (!data || !data.name) throw new Error('Invalid NFT data json');
+  console.log(`Minting ${nftCount} NFTs - ${data.name}`);
+  const res = await signingClient.mintNFTs(
+    account.address,
+    classId,
+    [...Array(nftCount).keys()].map(() => {
+      const id = `nft-${uuidv4()}`;
+      let { uri } = data;
+      const isUriHttp = uri && uri.startsWith('https://');
+      if (isUriHttp) uri = addParamToUrl(uri, { class_id: classId, nft_id: id });
+      return {
+        id,
+        uri,
+        metadata: {
+          name: data.name,
+          image: data.image,
+        },
+      };
+    }),
+  );
+  console.log(`Minting NFTs - Completed ${res.transactionHash}`);
+}
+
+function printHelp() {
+  console.log(`Usage:
+  MNEMONIC="...." node index.js --nft-count 100
+
+Required Paramters:
+  --nft-count: How many NFT to mint
+
+Optional Parameters:
+  --iscn-id: Use existing ISCN ID. If ISCN ID is not set, data in ./data/iscn.json will be used.
+  --class-id: Use existing NFT class ID.  If NFT class ID is not set, data in ./data/nft.json will be used.
+  --nft-max-supply: Define max supply for new NFT class
+  `);
+}
+
+async function run() {
+  if (!MNEMONIC) {
+    console.error('MNEMONIC is not defined in env!');
+    printHelp();
+    return;
+  }
+  const args = yargsParser(process.argv.slice(2));
+  const {
+    nftCount,
+    nftMaxSupply,
+    help,
+  } = args;
+  let {
+    iscnId,
+    classId,
+  } = args;
+  if (help) {
+    printHelp();
+    return;
+  }
+  if (!nftCount || !Number(nftCount)) {
+    console.error('Invalid NFT count');
+    printHelp();
+    return;
+  }
```

Comment:
Do we have to check the limitation on `nftCoint`?

Reply:
maybe we can check mintCount <= nftCount